data/random-teams.ts

```diff
@@ -1626,8 +1627,10 @@ export class RandomTeams {
 			const move = this.dex.moves.get(m);
 			if (move.damageCallback || move.damage) return true;
 			if (move.id === 'shellsidearm') return false;
-			// Magearna, though this can work well as a general rule
-			if (move.id === 'terablast' && moves.has('shiftgear')) return false;
+			// Magearna and doubles Dragonite, though these can work well as a general rule
+			if (
+				move.id === 'terablast' && (moves.has('shiftgear') || species.baseStats.atk >= species.baseStats.spa)
```

Comment:
```suggestion
				move.id === 'terablast' && (moves.has('shiftgear') || species.baseStats.atk > species.baseStats.spa)
```
Tera Blast defaults to Special if offensive stats are equal